src/main/java/com/osiris/autoplug/client/tasks/updater/search/CustomCheckURL.java

```diff
@@ -0,0 +1,76 @@
+package com.osiris.autoplug.client.tasks.updater.search;
+
+import com.google.gson.JsonObject;
+import com.osiris.autoplug.client.tasks.updater.plugins.MinecraftPlugin;
+import com.osiris.autoplug.client.tasks.updater.search.SearchResult;
+import com.osiris.autoplug.client.utils.UtilsURL;
+import com.osiris.jlib.json.Json;
+import com.osiris.jlib.logger.AL;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class CustomCheckURL {
+
+    public CustomCheckURL(){}
+
+    private boolean isInt(String s) {
+        try {
+            Integer.parseInt(s);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    public SearchResult doCustomCheck(MinecraftPlugin plugin) {
+
+        String url = plugin.getCustomCheckURL();
+        url = new UtilsURL().clean(url);
+        Exception exception = null;
+        String latest = null;
+        String type = ".jar";
+        String downloadUrl = null;
```

Comment:
Good job! 

One small thing, there should be at least an attempt further below to check if there is a download url provided in the response json, since all the user might provide is the custom check url and nothing else.

For example by looping through all keys and checking if a key contains download_url or asset_url or jar_url and then getting it's value.